test/cmd/newapp.sh

```diff
@@ -175,22 +175,22 @@ os::cmd::expect_failure_and_text "echo 'S P A C E S=test' | oc new-app php --env
 
 # new-build
 # check that env vars are not split on commas and warning is printed where they previously have
-os::cmd::expect_success_and_text 'oc new-build --binary php --env X=Y,Z=W -o yaml' 'value: Y,Z=W'
-os::cmd::expect_success_and_text 'oc new-build --binary php --env X=Y,Z=W -o yaml' 'no longer accepts comma-separated list'
-os::cmd::expect_success_and_text 'oc new-build --binary php --env X=Y,Z,W -o yaml' 'value: Y,Z,W'
-os::cmd::expect_success_and_not_text 'oc new-build --binary php --env X=Y,Z,W -o yaml' 'no longer accepts comma-separated list'
-os::cmd::expect_success_and_not_text 'oc new-build --binary php --env X=Y -o yaml' 'no longer accepts comma-separated list'
+os::cmd::expect_success_and_text 'oc new-build --binary php --to=binary --env X=Y,Z=W -o yaml' 'value: Y,Z=W'
+os::cmd::expect_success_and_text 'oc new-build --binary php --to=binary --env X=Y,Z=W -o yaml' 'no longer accepts comma-separated list'
+os::cmd::expect_success_and_text 'oc new-build --binary php --to=binary --env X=Y,Z,W -o yaml' 'value: Y,Z,W'
+os::cmd::expect_success_and_not_text 'oc new-build --binary php --to=binary --env X=Y,Z,W -o yaml' 'no longer accepts comma-separated list'
+os::cmd::expect_success_and_not_text 'oc new-build --binary php --to=binary --env X=Y -o yaml' 'no longer accepts comma-separated list'
```

Comment:
why did you have to add the --to=binary?

I would have expected that this command would pick up the input "php" image from the openshift namespace, and create a target output "php" in the current namespace, which should not be a circular reference.

Reply:
this centers around the whole circular dependency discussion we've been having .... the result is that with the php images present in the test's namespace (not the openshift namespace) is that php:latest has a from link to php:7.0, and with them pointing to the same image, the current logic that compares the docker image ref fails

when this test runs, there *ARE NO IMAGESTREAMS* in the openshift namespace

the prior incarnation did not fail because we were not reusing the real image stream, but doing the circular check against our newly created image stream that was empty

Reply:
ok, if there is an existing php imagestream (with a 7.0 and latest tag as you describe, and the update behavior you confirmed) then yeah, it should have been caught as a circular reference before and i agree w/ adding the --to here.

that said, i'm not seeing that behavior...  i created a php tag with a 7.0 and a latest that follows 7.0, then i did a build that pushed to latest and it did not update the 7.0 tag:

latest tag after push to latest tag:
```
- items:
    - created: 2017-10-17T14:24:23Z
      dockerImageReference: 172.30.66.174:5000/p3/php@sha256:0d3de95379d5fb1192e8210886fe8b6bc082c3693da3504bd39b2bc51bc17b85
      generation: 2
      image: sha256:0d3de95379d5fb1192e8210886fe8b6bc082c3693da3504bd39b2bc51bc17b85
    - created: 2017-10-17T14:22:19Z
      dockerImageReference: centos/php-70-centos7@sha256:e00e498810997b855ae23f1531d2e2df490beab37c790a5a50587df4bfcb15e1
      generation: 2
      image: sha256:e00e498810997b855ae23f1531d2e2df490beab37c790a5a50587df4bfcb15e1
    tag: latest
```

7.0 tag after push to latest tag:
```
    tag: "7.0"
  - items:
    - created: 2017-10-17T14:22:19Z
      dockerImageReference: centos/php-56-centos7@sha256:7c303efa89bc3513e0da6028b86a8954ab5836706988bc257ffe0d16ae2374bc
      generation: 2
      image: sha256:7c303efa89bc3513e0da6028b86a8954ab5836706988bc257ffe0d16ae2374bc
```

Reply:
(and i don't think the latest tag will get updated on a push to 7.0 either.  I think you'd have to push to 7.0 and then explicitly do an import-image)